web-rdfs/include/web_rdfs_server.h

```diff
@@ -0,0 +1,136 @@
+
+using HttpServer = SimpleWeb::Server<SimpleWeb::HTTP>;
+
+// the .proto file implementation's namespace, used for messages
+using hadoop::hdfs::GetFileInfoRequestProto;
+using hadoop::hdfs::GetFileInfoResponseProto;
+using hadoop::hdfs::MkdirsRequestProto;
+using hadoop::hdfs::MkdirsResponseProto;
+using hadoop::hdfs::DeleteRequestProto;
+using hadoop::hdfs::DeleteResponseProto;
+using hadoop::hdfs::CreateRequestProto;
+using hadoop::hdfs::CreateResponseProto;
+using hadoop::hdfs::GetBlockLocationsRequestProto;
+using hadoop::hdfs::GetBlockLocationsResponseProto;
+using hadoop::hdfs::GetServerDefaultsRequestProto;
+using hadoop::hdfs::GetServerDefaultsResponseProto;
+using hadoop::hdfs::FsServerDefaultsProto;
+using hadoop::hdfs::RenewLeaseRequestProto;
+using hadoop::hdfs::RenewLeaseResponseProto;
+using hadoop::hdfs::CompleteRequestProto;
+using hadoop::hdfs::CompleteResponseProto;
+using hadoop::hdfs::AbandonBlockRequestProto;
+using hadoop::hdfs::AbandonBlockResponseProto;
+using hadoop::hdfs::AddBlockRequestProto;
+using hadoop::hdfs::AddBlockResponseProto;
+using hadoop::hdfs::RenameRequestProto;
+using hadoop::hdfs::RenameResponseProto;
+using hadoop::hdfs::SetPermissionResponseProto;
+using hadoop::hdfs::GetListingRequestProto;
+using hadoop::hdfs::GetListingResponseProto;
+using hadoop::hdfs::SetReplicationResponseProto;
+using hadoop::hdfs::GetEZForPathResponseProto;
+using hadoop::hdfs::SetOwnerResponseProto;
+using hadoop::hdfs::GetContentSummaryRequestProto;
+using hadoop::hdfs::GetContentSummaryResponseProto;
+using hadoop::hdfs::RecoverLeaseRequestProto;
+using hadoop::hdfs::RecoverLeaseResponseProto;
+using hadoop::hdfs::Rename2RequestProto;
+using hadoop::hdfs::Rename2ResponseProto;
+using hadoop::hdfs::AppendRequestProto;
+using hadoop::hdfs::ConcatRequestProto;
+using hadoop::hdfs::SetReplicationRequestProto;
+using hadoop::hdfs::GetEZForPathRequestProto;
+using hadoop::hdfs::SetOwnerRequestProto;
+using hadoop::hdfs::SetPermissionRequestProto;
+
+class WebRDFSServer {
+ public:
+  /**
+   * Create an WebRDFSServer instance.
+   *
+   * @param port Port number on which to listen for HTTP requests.
+   */
+  explicit WebRDFSServer(int16_t port, zkclient::ZkNnClient *zk_arg);
+
+  /**
+   * Start the WebRDFS server.
+   */
+  void start();
+
+ private:
+  /**
+   * Simple-Web-Server HTTP server instance.
+   */
+  HttpServer server;
+
+  /**
+   * client to communicate with zookeeper
+   */
+  zkclient::ZkNnClient *zk;
+
+  /**
+   * Register an endpoint handler on the server.
+   *
+   * @param pattern URL pattern to register.
+   * @param verb HTTP verb to register.
+   * @param handler Handler callback function associated with this endpoint.
+   */
+  void register_handler(
+    std::string pattern,
+    const char verb[],
+    std::function<void(
+      std::shared_ptr<HttpServer::Response> response,
+      std::shared_ptr<HttpServer::Request> request)> handler);
+
+
+  std::string getFileInfo(GetFileInfoRequestProto req);
```

Comment:
Why return std::string if all functions return ""? Should these perhaps pass along error information from zk?